### PR TITLE
This commit includes task 7

### PR DESCRIPTION
**Pull request by Wainhouse**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -54,104 +55,140 @@ describe("GET /api/topics", () => {
...
+      });
+  });
+  it("400: Invalid article_id", () => {
+    return request(app)
+      .get("/api/articles/sdfsdfsdf/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("400: Invalid article_id");
+      });
+  });
```
Comment (NRMSMITH) [inline __tests__/app.test.js:193]: Excellent testing so far - consider these instances:
201 - ignore unnecessary properties in the sent object
400 - missing required fields (i.e. no body or no username)
404 - username does not exist

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Need to add a few more tests but your code looks clean 👍 please submit to this pr and send us a nchelp pr when you've added the tests so we can check them :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -54,104 +56,182 @@ describe("GET /api/topics", () => {
...
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.comment.body).toBe("Yo wadup man!");
+        expect(body.comment.author).toBe("butter_bridge");
+        expect(body.comment.votes).toBe(0);
+        expect(typeof body.comment).toBe("object");
+        expect(body.comment).toHaveProperty("comment_id");
+        expect(body.comment).toHaveProperty("votes");
+        expect(body.comment).toHaveProperty("created_at");
+        expect(body.comment.comment_id).toBeGreaterThan(0);
```
Comment (itsaliabdallah) [inline __tests__/app.test.js:176]: nice test but some of these, are duplicated tests.

File: api/controllers/articles.controller.js [hunk 3]
```diff
@@ -1,31 +1,55 @@
...
+    .catch((err) => {
+      next(err);
+    });
+};
+
+exports.postComment = (req, res, next) => {
+  const comment = req.body;
+  const articleId = req.params.article_id;
+  if(!comment.body || !comment.username) {
+    return res.status(400).send({msg: `400: not found, make sure you have included a username and a comment` })
```
Comment (itsaliabdallah) [inline api/controllers/articles.controller.js:42]: nice

**Pull request merged**